Content.Shared/Doors/Systems/SharedDoorSystem.cs

```diff
@@ -396,6 +396,25 @@ public void OnPartialOpen(EntityUid uid, DoorComponent? door = null)
         Dirty(uid, door);
 
     }
+
+    /// <summary>
+    /// Opens and then bolts a door.
+    /// Different from emagging this does not remove the access reader, so it can be repaired by simply unbolting the door.
+    /// </summary>
+    public bool TryOpenAndBolt(EntityUid uid, DoorComponent? door = null, AirlockComponent? airlock = null)
```

Comment:
i suppose you could realize emagdoor method via this new one

Reply:
No you can't. In the emag version the check for being unbolted and powered happens in a separate event, so that it does not consume a charge if it fails.